### PR TITLE
Fix plAnimPath zero init

### PR DESCRIPTION
**Pull request by Hoikas**
Fixes random heap garbage being written to the disk, unless you feel like being explicit.

### CODE REVIEW

File: core/PRP/Animation/plAnimPath.h [hunk 1]
```diff
@@ -45,7 +45,7 @@ class PLASMA_DLL plAnimPath : public plCreatable {
    hsAffineParts fParts;
 
 public:
-    plAnimPath() : fController(NULL), fTMController(NULL) { }
+    plAnimPath() : fController(), fTMController(), fAnimPathFlags(), fMinDistSq(), fLength() { }
```
Comment (zrax) [inline core/PRP/Animation/plAnimPath.h]: These should be reordered to match declaration order (else GCC warnings)

Comment (Hoikas) [reply core/PRP/Animation/plAnimPath.h]: 😨  what's GCC smokin'?

**Comment by Hoikas** [comment]
Updated to work around GCC's substance dependency issues.

**Comment by zrax** [comment]
It may seem like GCC is just being picky, but keep in mind that the compiler will call the member initializers in the order they're declared in the class, _not_ in the order they're listed in the constructor.  So it does make some sense for the occasional case where that difference matters...

**Pull request merged**